File: app/src/main/java/com/github/ljarka/bakingapp/detail/IngredientsAdapter.java
package com.github.ljarka.bakingapp.detail;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.github.ljarka.bakingapp.R;
import com.github.ljarka.bakingapp.network.model.IngredientData;

import java.util.Collections;
import java.util.List;

public class IngredientsAdapter extends RecyclerView.Adapter<IngredientsAdapter.IngredientsViewHolder> {

    private List<IngredientData> ingredients = Collections.emptyList();

    @Override
    public IngredientsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new IngredientsViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.item_ingredient, parent, false));
    }

    @Override
    public void onBindViewHolder(IngredientsViewHolder holder, int position) {
        holder.name.setText(ingredients.get(position).getIngredient());
    }

    @Override
    public int getItemCount() {
        return ingredients.size();
    }

    void setIngredients(List<IngredientData> ingredients) {
        this.ingredients = ingredients;
        notifyDataSetChanged();
    }

    class IngredientsViewHolder extends RecyclerView.ViewHolder {
        TextView name;

        IngredientsViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
        }
    }
}

File: app/src/main/java/com/github/ljarka/bakingapp/detail/IngredientsAdapter.kt
package com.github.ljarka.bakingapp.detail

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.github.ljarka.bakingapp.R
import com.github.ljarka.bakingapp.network.model.IngredientData
import kotlinx.android.synthetic.main.item_ingredient.view.*

class IngredientsAdapter : RecyclerView.Adapter<IngredientsAdapter.IngredientsViewHolder>() {
    var ingredients: List<IngredientData>? = null
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): IngredientsViewHolder {
        return IngredientsViewHolder(LayoutInflater.from(parent?.context)
                .inflate(R.layout.item_ingredient, parent, false))
    }

    override fun onBindViewHolder(holder: IngredientsViewHolder?, position: Int) {
        holder?.itemView?.name?.text = ingredients?.get(position)?.ingredient
    }

    override fun getItemCount(): Int {
        return ingredients?.size ?: 0
    }

    class IngredientsViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView)
}